src/main/java/co/cask/bigquery/BigQuerySink.java

```diff
@@ -120,26 +122,27 @@ public void prepareRun(BatchSinkContext context) throws Exception {
       String name = field.getName();
       Schema.Type type = field.getSchema().getType();
 
-      if (type == Schema.Type.RECORD) {
-        continue;
-      }
-
-      String typeName = "STRING";
+      String typeName;
+      Schema fieldSchema = field.getSchema();
       if (type.isSimpleType()) {
-        typeName = simpleType(field.getSchema());
+        typeName = getTableDataType(fieldSchema);
       } else if (type == Schema.Type.UNION) {
-        typeName = unionType(field.getSchema());
-      } else if (type == Schema.Type.RECORD) {
-        throw new IllegalArgumentException(
-          String.format("Current implementation of BigQuery sink does not yet support complex records.")
-        );
+        fieldSchema =  fieldSchema.isNullable()? fieldSchema.getNonNullable() : fieldSchema;
```

Comment:
this should replace the existing `if (type.isSimpleType())`. There should be no union type check.